compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -149,15 +149,15 @@ class SyntheticMethods(thisTransformer: DenotTransformer) {
      *  gets the hashCode method:
      *
      *    def hashCode: Int = {
-     *      <synthetic> var acc: Int = 0xcafebabe;
+     *      <synthetic> var acc: Int = productPrefix.hashCode()
      *      acc = Statics.mix(acc, x);
      *      acc = Statics.mix(acc, Statics.this.anyHash(y));
      *      Statics.finalizeHash(acc, 2)
      *    }
      */
     def caseHashCodeBody(implicit ctx: Context): Tree = {
       val acc = ctx.newSymbol(ctx.owner, "acc".toTermName, Mutable | Synthetic, defn.IntType, coord = ctx.owner.pos)
-      val accDef = ValDef(acc, Literal(Constant(0xcafebabe)))
+      val accDef = ValDef(acc, This(clazz).select(defn.Product_productPrefix).select(defn.Any_hashCode).ensureApplied)
```

Comment:
Given that this code generates also `productPrefix` itself, why not inline constant `ownName` here instead?